app/views/contexts/summary_context.py

```diff
@@ -4,48 +4,119 @@
-            if section.get("summary", {}).get("items"):
-                break
-
-            for group in section_summary_context(return_to=return_to)["summary"][
-                "groups"
-            ]:
-                yield group
+            if repeat := self._schema.get_repeat_for_section(section_id):
+                for_repeat = self._list_store[repeat["for_list"]]
+
+                if for_repeat.count > 0:
+                    for item in for_repeat.items:
+                        location = Location(
+                            section_id=section_id,
+                            list_name=for_repeat.name,
+                            list_item_id=item,
+                        )
+                        section_summary_context = SectionSummaryContext(
+                            language=self._language,
+                            schema=self._schema,
+                            answer_store=self._answer_store,
+                            list_store=self._list_store,
+                            progress_store=self._progress_store,
+                            metadata=self._metadata,
+                            response_metadata=self._response_metadata,
+                            current_location=location,
+                            routing_path=self._router.routing_path(
+                                section_id=section_id, list_item_id=item
+                            ),
+                        )
+
+                        yield from section_summary_context.build_summary(
+                            return_to=return_to,
+                            get_refactored_groups=False,
+                            summary_type="Summary",
+                            view_submitted_response=self.view_submitted_response,
+                        ).get("groups", [])
+            else:
+                location = Location(section_id=section_id, list_item_id=None)
+                section_summary_context = SectionSummaryContext(
+                    language=self._language,
+                    schema=self._schema,
+                    answer_store=self._answer_store,
+                    list_store=self._list_store,
+                    progress_store=self._progress_store,
+                    metadata=self._metadata,
+                    response_metadata=self._response_metadata,
+                    current_location=location,
+                    routing_path=self._router.routing_path(section_id),
+                )
+
+                yield from section_summary_context.build_summary(
+                    return_to=return_to,
+                    get_refactored_groups=False,
+                    summary_type="Summary",
+                    view_submitted_response=self.view_submitted_response,
+                ).get("groups", [])
+
+
+def set_unique_group_ids(groups: list[ImmutableDict]) -> list[ImmutableDict]:
+    checked_ids = set()
+    id_value = 0
+
+    for group in groups:
+        if "id" in group:
```

Comment:
Groups always have id? from validator: `"required": ["id", "blocks"]`

Reply:
Removed the if 👍